app/src/main/java/dev/arkbuilders/navigator/data/utils/PathExt.kt

```diff
@@ -0,0 +1,58 @@
+package dev.arkbuilders.navigator.data.utils
+
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.exists
+import kotlin.io.path.extension
+import kotlin.io.path.nameWithoutExtension
+import kotlin.io.path.notExists
+
+val ROOT_PATH: Path = Paths.get("/")
+
+val ANDROID_DIRECTORY: Path = Paths.get("Android")
+
+fun Path.findNotExistCopyName(name: Path): Path {
+    val originalNamePath = this.resolve(name.fileName)
+    if (originalNamePath.notExists())
+        return originalNamePath
+
+    var filesCounter = 1
+
+    val formatNameWithCounter =
+        "${name.nameWithoutExtension}_$filesCounter.${name.extension}"
+
+    var newPath = this.resolve(formatNameWithCounter)
+
+    while (newPath.exists()) {
+        newPath = this.resolve(formatNameWithCounter)
+        filesCounter++
+    }
+    return newPath
+}
+
+fun findLongestCommonPrefix(paths: List<Path>): Path {
+    if (paths.isEmpty()) {
+        throw IllegalArgumentException(
+            "Can't search for common prefix among empty collection"
+        )
+    }
+
+    if (paths.size == 1) {
+        return paths.first()
+    }
+
+    return tailrec(ROOT_PATH, paths).first
+}
+
+private fun tailrec(_prefix: Path, paths: List<Path>): Pair<Path, List<Path>> {
```

Comment:
We need another name for `_prefix`... Doesn't `_` usually mean that the variable isn't actually used?

Reply:
Argument name should start with lower case character. I updated it and also rename others